pkg/safetime/safetime.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safetime
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// TimeSinceSafe returns the duration since t. If the duration is negative,
+// returns false to indicate the fact.
+//
+// Used to workaround a malfunctioning monotonic clock.
+func TimeSinceSafe(t time.Time, logger *log.Entry) (bool, time.Duration) {
```

Comment:
It's not too important but it might be nice to match the normal go `ok` variable order, and have the boolean be second (to match `foo, ok := myMap[myKey]` and `bar, ok := myIfc.(myConcreteType)`)